packages/appcd-nodejs/test/test-nodejs.js

```diff
@@ -333,10 +335,67 @@ describe('nodejs', () => {
 				});
 		});
 
-		it('should error if arch is invalid', () => {
+		it('should error if arch is invalid', done => {
+			spawnNode({ arch: 'foo' })
+				.then(() => {
+					throw new Error('Expected exception to be thrown');
+				}, err => {
+					expect(err.message).to.equal('Expected arch to be "x86" or "x64"');
+					done();
+				})
+				.catch(done);
+		});
+	});
+
+	describe('purgeUnusedNodejsExecutables()', () => {
+		it('should find all unused Node.js executables', () => {
+			const nodeExecutable = getNodeFilename();
+			const now = Date.now();
+			const { platform } = process;
+			const tmpDir = makeTempDir();
+
+			const createNodeExecutable = (dir, lastrun) => {
+				fs.mkdirsSync(dir);
+				fs.writeFileSync(path.join(dir, nodeExecutable), '# mock node executable');
+				if (lastrun) {
+					fs.writeFileSync(path.join(dir, '.lastrun'), lastrun);
+				}
+			};
+
+			createNodeExecutable(path.join(tmpDir, 'v8.9.1', platform, 'x64'));
+			createNodeExecutable(path.join(tmpDir, 'v8.9.2', platform, 'x64'), now - (24 * 60 * 60 * 1000)); // 1 day
+			createNodeExecutable(path.join(tmpDir, 'v8.9.3', platform, 'x64'), now);
+
+			const results = purgeUnusedNodejsExecutables({
+				maxAge: 60 * 1000, // 1 minute
+				nodeHome: tmpDir
+			});
+
+			expect(results).to.deep.equal([
+				{
+					arch: 'x64',
+					platform: 'darwin',
```

Comment:
`expect(results).to.deep.equal([
				{
					arch: 'x64',
					platform,
					version: 'v8.9.2'
				}
			]);
`

(See jenkins failure on Linux)